src/Maestro/Microsoft.DotNet.Maestro.Tasks.Tests/CreateMergedManifestBuildModelTests.cs

```diff
@@ -0,0 +1,289 @@
+        private static readonly string sourceBranch = "thisIsASourceBranch";
+        private static readonly string commitSourceVersion = "thisIsASourceVersion";
+
+        private PushMetadataToBuildAssetRegistry pushMetadata;
+        private BuildModel expectedBuildModel;
+
+        private AssetData assetDataWithoutName = new AssetData(true)
+        { Version = "noNameVersion" };
+
+        private AssetData assetDataWithoutVersion = new AssetData(true)
+        { Name = "noVersionData" };
+
+        private AssetData nonShippingAssetData = new AssetData(true)
+        {
+            Name = "NonShippingAssetData",
+            Version = "nonShippingAsssetVersion"
+        };
+
+        private AssetData shippingAssetData = new AssetData(false)
+        {
+            Name = "ShippingAssetData",
+            Version = "shippingAssetVersion"
+        };
+
+        private ManifestBuildData manifestBuildData = new ManifestBuildData(
+            new Manifest
+            {
+                InitialAssetsLocation = initialAssetsLocation,
+                AzureDevOpsBuildId = azDoBuildId,
+                AzureDevOpsBuildDefinitionId = azDoBuildDefId,
+                AzureDevOpsAccount = azDoAccount,
+                AzureDevOpsProject = azDoProject,
+                AzureDevOpsBuildNumber = azDoBuildNumber,
+                AzureDevOpsRepository = azDoRepo,
+                AzureDevOpsBranch = azDoBranch,
+                PublishingVersion = publishingVersion,
+                IsReleaseOnlyPackageVersion = isReleasePackage
+            });
+
+        [SetUp]
+        public void TestCaseSetup()
+        {
+            Mock<IGetEnvProxy> getEnvMock = new Mock<IGetEnvProxy>();
+            getEnvMock.Setup(a => a.GetEnv("BUILD_REPOSITORY_NAME")).Returns(buildRepoName);
+            getEnvMock.Setup(b => b.GetEnv("BUILD_BUILDNUMBER")).Returns(buildNumber);
+            getEnvMock.Setup(c => c.GetEnv("BUILD_SOURCEBRANCH")).Returns(sourceBranch);
+            getEnvMock.Setup(d => d.GetEnv("BUILD_SOURCEVERSION")).Returns(commitSourceVersion);
+            getEnvMock.SetReturnsDefault("MissingEnvVariableCheck!");
+
+            pushMetadata = new PushMetadataToBuildAssetRegistry
+            {
+                getEnvProxy = getEnvMock.Object
+            };
+
+            expectedBuildModel = new BuildModel(
+            new BuildIdentity
+            {
+                Attributes = manifestBuildData.ToDictionary(),
+                Name = buildRepoName,
+                BuildId = buildNumber,
+                Branch = sourceBranch,
+                Commit = commitSourceVersion,
+                IsStable = false,
+                PublishingVersion = (PublishingInfraVersion)manifestBuildData.PublishingVersion,
+                IsReleaseOnlyPackageVersion = bool.Parse(isReleasePackage)
+            });
+        }
+
+        [Test]
+        [Ignore("Fails due to bug https://github.com/dotnet/arcade/issues/6677")]
```

Comment:
We should probably fix this issue first. Adding a bunch of test that are disabled is a little odd...

Reply:
I was attempting to avoid code rot & hoping I could just squeeze in the other change before the review was done on this one (might have worked if I hadn't gotten sick). I'd prefer to get it in since I don't think we're likely to do an arcade release before the end of the year, but I'm happy to wait if I can do it this year.

Reply:
The split of this behavior between arcade and arcade-services is so weird. :-( It's probably fine.